HotFix/Core/FIXMessageWriter.cs

```diff
@@ -8,123 +8,159 @@ public class FIXMessageWriter
         private const byte SOH = (byte)'\u0001';
         private const byte EQL = (byte) '=';
 
-        internal readonly byte[] _buffer;
-        internal int _end;
-
-        private readonly int _headerEnd;
+        private readonly byte[] _body;
         private int _bodyEnd;
-        private int _trailerEnd;
 
+        internal readonly byte[] Buffer;
+        internal int End;
```

Comment:
As discussed earlier, it might make more sense to call this `Length` as _end_ doesn't mean much from outside...